### PR TITLE
Replace deprecated "removeEventListener" syntax

### PR DESCRIPTION
**Pull request by adrienharnay**
This syntax has been deprecated for over a year now. This PR addresses this.

### CODE REVIEW

File: CodePush.js [hunk 1]
```diff
@@ -209,22 +209,23 @@ async function tryReportStatus(statusReport, resumeListener) {
...
         if (refreshedStatusReport) {
-          tryReportStatus(refreshedStatusReport, resumeListener);
+          tryReportStatus(refreshedStatusReport, resumeListener, localAppStateListener);
         } else {
-          AppState.removeEventListener("change", resumeListener);
+          localAppStateListener && localAppStateListener.remove();
         }
       };
-      AppState.addEventListener("change", resumeListener);
+      localAppStateListener = AppState.addEventListener("change", resumeListener);
```
Comment (thewulf7) [inline CodePush.js:228]: Not sure I understand this line. Basically token (aka localAppStateListener) from adding listener, as later on passed to scope of resumeListener right to remove it?

If so Im not sure I understand what line 216 is doing.

Comment (adrienharnay) [reply CodePush.js:228]: Hello, thank you for your answer. What I wanted to do:
- initialize `localAppStateListener` with the `appStateListener ` parameter l.216
- assign it the return of `addEventListener` l.228
- in the event listener, access `localAppStateListener` which will hold the return from `addEventListener`

I agree that initializing `localAppStateListener` might be pointless. Do you want me to make this change or did you have something else in mind?

**Comment by jbaudanza** [comment]
This function has been removed from React Native 0.70, so this warning is now an error:

```
c.AppState.removeEventListener is not a function. (In 'c.AppState.removeEventListener("change",t)', 'c.AppState.removeEventListener' is undefined)
```

**Review by trcoffman — CHANGES_REQUESTED** [review]
Upon reading this PR, I authored #2385 which I think is a cleaner fix. There's no reason to introduce a 3rd argument, as the only purpose of the argument is to (a) determine whether or not we have already established a resumeListener, and (b) to remove the listener at the appropriate time. When the function recurses, it only needs the subscription, not the closure itself, so rather than introduce another argument, we just change what the argument is. This PR can be closed and replaced by #2385.

**Comment by Titozzz** [comment]
@thewulf7 Gentle nudge, feel free to merge #2385 and close this one if the syntax seems clearer to you

**Comment by thewulf7** [comment]
Merged #2385.Closing this

**Pull request closed without merging**